src/components/ShoppingList/ShoppingList.js

```diff
@@ -21,13 +28,24 @@ function ShoppingList({ listId }) {
         </div>
       )}
 
-      {!loading && listItems && (
+      {/* !loading is required to access listItems array */}
+      {/* listItems.empty is a Boolean for whether querySnapshot is empty */}
+      {!loading && !listItems.empty && (
         <ul className="shopping-list__list">
           {listItems.docs.map((doc) => (
             <ShoppingListItem key={doc.id} item={doc.data()} />
           ))}
         </ul>
       )}
+
+      {!loading && listItems.empty && (
```

Comment:
Given that you need to check for `!loading` in both options here I wanted to make a suggestion for a possible refactor to increase the readability.

You can check if `!loading` is true and if so call to a function that will do any additional checks and processing needed and return the appropriate elements to render.  This separates the concerns in a way that makes it easier to quickly scan this return statement and get a sense that it looks for 3 possible states and then renders something based on those states. 

If I were another developer working with this code and  I needed to see more closely what is returned if `!loading` is true then I can look to the function for that information, but if that is not a concern for me I can move on.

That would look something like this:
```js
{!loading && createListElement()}
```

and then:
```js
const createListElement = () => {
    if (listItems.empty) {
      return (
        <>
          <p>Your shopping list is currently empty.</p>
          <button className="button" type="button" onClick={handleClick}>
            Add Item
          </button>
        </>
      );
    }

    return (
      <ul className="shopping-list__list">
        {listItems.docs.map((doc) => (
          <ShoppingListItem key={doc.id} item={doc.data()} />
        ))}
      </ul>
    );
  };
  ``` 

You likely wouldn't need the explanatory comment above with this approach either as it is a bit easier to follow what is happening through the code alone.

Reply:
This is a great suggestion. I've added a commit for this. Thanks April!